src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,509 @@
+
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.buildings.AbstractBuilding;
+import com.minecolonies.colony.buildings.AbstractBuildingWorker;
+import com.minecolonies.colony.buildings.BuildingGuardTower;
+import com.minecolonies.colony.jobs.JobGuard;
+import com.minecolonies.colony.permissions.Permissions;
+import com.minecolonies.entity.ai.basic.AbstractEntityAISkill;
+import com.minecolonies.entity.ai.util.AIState;
+import com.minecolonies.entity.ai.util.AITarget;
+import com.minecolonies.tileentities.TileEntityColonyBuilding;
+import com.minecolonies.util.BlockPosUtil;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.item.EntityItem;
+import net.minecraft.entity.monster.EntityMob;
+import net.minecraft.entity.monster.EntitySlime;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.*;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.entity.ai.util.AIState.*;
+
+/**
+ * Abstract class which contains all the guard basics let it be range, melee or magic.
+ */
+public abstract class AbstractEntityAIGuard extends AbstractEntityAISkill<JobGuard>
+{
+    /**
+     * The start search distance of the guard to track/attack entities may get more depending on the level.
+     */
+    private static final double MAX_ATTACK_DISTANCE = 20.0D;
+
+    /**
+     * Distance the guard starts searching.
+     */
+    private static final int START_SEARCH_DISTANCE = 5;
+
+    /**
+     * Basic delay after operations.
+     */
+    private static final int BASE_DELAY = 1;
+
+    /**
+     * Max amount the guard can shoot arrows before restocking.
+     */
+    private static final int MAX_ATTACKS = 50;
+
+    /**
+     * Y range in which the guard detects other entities.
+     */
+    private static final double HEIGHT_DETECTION_RANGE = 10D;
+
+    /**
+     * Path that close to the patrol target.
+     */
+    private static final int PATH_CLOSE      = 3;
+
+    /**
+     * Worker gets this distance times building level away from his building to patrol.
+     */
+    private static final int PATROL_DISTANCE = 20;
+
+    /**
+     * Horizontal range in which the guard picks up items
+     */
+    private static final double RANGE_HORIZONTAL_PICKUP = 20.0D;
+    /**
+     * Vertical range in which the guard picks up items
+     */
+    private static final double RANGE_VERTICAL_PICKUP   = 2.0D;
+
+    /**
+     * Amount of ticks after which the guard stops trying to gather and tries to get onto another task.
+     */
+    private static final int   STUCK_WAIT_TICKS        = 20;
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)